src/main/java/core/basesyntax/operation/SupplyOperation.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax.operation;
+
+import java.util.Map;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void execute(Map.Entry<String, Integer> entry, int value) {
+        entry.setValue(entry.getValue() + value);
+    }
```

Comment:
The `SupplyOperation` class should validate the input parameters to ensure they are not null and the value is not negative, as supplying a negative value would not be a valid operation.